hardware_interface/src/lexical_casts.cpp

```diff
@@ -29,7 +31,19 @@ double stod(const std::string & s)
   }
   return result;
 }
+#else
+double stod(const std::string & text)
+{
+  double result_value;
+  const auto parse_result = std::from_chars(text.data(), text.data() + text.size(), result_value);
+  if (parse_result.ec == std::errc())
+  {
+    return result_value;
+  }
 
+  return 0.0;
+}
```

Comment:
The alternative implementation of `stod()` does not conform with the contract because it returns 0 instead of throwing `std::invalid_argument` if string parsing fails.

Reply:
invalid_argument sounds reasonable to me. I agree with you that we shouldn't return 0.0 by default unless there is a default argument parsed to the method.

Reply:
same thought. invalid_argument looks better. Let me give it a try.